src/shared/lib/ExtendedRouterStore.ts

```diff
@@ -2,19 +2,16 @@ import { RouterStore } from 'mobx-react-router';
 import { action, computed } from 'mobx';
 import * as _ from 'lodash';
 import * as $ from 'jquery';
+import URI from 'urijs';
 
 // NOTE: there's a problem with mobx-react-router types so we have to type a few things here
 // which should be derived from typings of super class
 export default class ExtendedRouterStore extends (RouterStore as FunctionConstructor) {
 
     @computed get query(): {[key: string] : string} {
-        // var route = new RT(stores.routing.location.pathname);
-        var queryString: { [key: string] : string } = {};
-        decodeURIComponent(this.location.search).replace(
-            new RegExp("([^?=&]+)(=([^&]*))?", "g"),
-            function($0, $1, $2, $3) { queryString[$1] = $3; return $0}
-        )
-        return queryString;
+
+        return (URI.parseQuery(this.location.search) as {[key: string] : string});
```

Comment:
A note, we already have at least 2 other URL building/parsing libraries installed.  So, at some point we should consolidate.  I will make a card for that.